source/Calamari.AzureAppService/ArmClientExtensions.cs

```diff
@@ -74,6 +74,22 @@ public static async Task<Stream> GetPublishingProfileXmlWithSecrets(this ArmClie
                    };
         }
 
+        public static async Task<bool> IsScmPublishEnable(this ArmClient armClient, AzureTargetSite targetSite)
+        {
+            if (targetSite.HasSlot)
+            {
+                var webSiteSlotResource = armClient.GetWebSiteSlotResource(targetSite.CreateResourceIdentifier());
+                var scmSiteBasicPublishingCredentialsPolicy = await webSiteSlotResource.GetScmSiteSlotBasicPublishingCredentialsPolicy().GetAsync();
+                return scmSiteBasicPublishingCredentialsPolicy.Value.Data.Allow ?? false;
```

Comment:
I doubt there's any scenario where this would fail or return without a value, but at the moment we depend on SCM being enabled, would it make more sense to keep the existing deployment behaviour as a default and fall-back to true here. Just a though, if there's a reason for defaulting to false all good.